src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java

```diff
@@ -89,7 +107,10 @@ LEFT JOIN (SELECT DISTINCT dataset_id AS id FROM dar_dataset) dar_ds_ids ON dar_
         LEFT JOIN dataset_property dp ON dp.dataset_id = d.dataset_id
         LEFT JOIN dictionary k ON k.key_id = dp.property_key
         LEFT JOIN consent_associations ca ON ca.dataset_id = d.dataset_id
-        LEFT JOIN file_storage_object fso ON fso.entity_id = d.dataset_id::text AND fso.deleted = false
+        LEFT JOIN study s ON s.study_id = d.study_id
+        LEFT JOIN study_property sp ON sp.study_id = s.study_id
+        LEFT JOIN dataset s_dataset ON s_dataset.study_id = s.study_id
+        LEFT JOIN file_storage_object fso ON (fso.entity_id = d.dataset_id::text OR fso.entity_id = s.uuid::text) AND fso.deleted = false
```

Comment:
I noticed in this PR that FSO entity ids for datasets are linked to `dataset_id` and not some UUID. This means if we linked to another table's `id` we could MAYBE run into collisions - however, I am assuming that all FSO categories will be associated which just one kind of entity - for example, `ALTERNATIVE_DATA_SHARING_PLAN`s will only be associated with Studies, etc. So, I don't think it's a big deal - just something that I noticed as a yellow-flag.

Reply:
One thing that we're not quite addressing yet is how to format a Dataset + Study where there are overlapping properties. For example, if a Study has a PI name and a Dataset has a PI name, which one do we show? Right now, there are none in the study table but as soon as we get one, we'll need to sort out what to do since the UI is expecting a suite of dataset properties with specific types to exist. I see two areas where we'll need work - the standard dataset catalog page and the study display page.